modules/solid_mechanics/include/bcs/ExplicitDirichletBCBase.h

```diff
@@ -41,4 +43,25 @@ class DirectDirichletBCBase : public NodalBC
 
   const Real & _u_old;
   const Real & _u_dot_old;
+
+  /// explicit time integrator
+  const ExplicitMixedOrder * _exp_integrator;
+
+  /// variable time order need for computing BC
+  ExplicitMixedOrder::TimeOrder _var_time_order;
+
+private:
+  /**
+   * Initialize the lumped mass matrix.
+   * @return lumped mass matrix vector
+   */
+  const NumericVector<Number> & initMassDiag()
+  {
+    const auto & nl = _fe_problem.getNonlinearSystemBase(_sys.number());
+    if (nl.hasVector("mass_matrix_diag_inverted"))
+      return nl.getVector("mass_matrix_diag_inverted");
+
+    mooseError("Lumped mass matrix is missing. Make sure ExplicitMixedOrder is being used as the "
+               "time integrator.");
+  }
```

Comment:
I suggest move the implementation into the source file.

I don't like the name referencing "diag". It's lumped, which is different from taking the diagonal. I think this is confusing.